cupy/_math/misc.py

```diff
@@ -399,7 +399,24 @@ def nan_to_num(x, copy=True, nan=0.0, posinf=None, neginf=None):
     return _nan_to_num(x, nan, posinf, neginf, out=out)
 
 
-# TODO(okuta): Implement real_if_close
+def real_if_close(x, tol=100):
+    """If input is complex with all imaginary parts close to zero, return real
+    parts.
+    “Close to zero” is defined as tol * (machine epsilon of the type for x).
+
+    .. seealso:: :func:`numpy.real_if_close`
+
+    """
+    x = cupy.asanyarray(x)
+    if not issubclass(x.dtype.type, cupy.complexfloating):
+        return x
+    if tol > 1:
+        from numpy.core import getlimits
+        f = getlimits.finfo(x.dtype.type)
```

Comment:
```suggestion
        f = numpy.finfo(x.dtype.type)
```
`numpy.core.getlimits` is an undocumented API.